Sources/SwiftDocC/Indexing/Navigator/NavigatorIndex+Ext.swift

```diff
@@ -78,12 +78,18 @@ extension RenderNode {
     
     /// Returns a navigator title preferring the fragments inside the metadata, if applicable.
     func navigatorTitle() -> String? {
-        let pageType = navigatorPageType()
-        guard ![.framework, .class, .structure, .enumeration, .protocol, .typeAlias, .associatedType].contains(pageType) else { return metadata.title }
-        guard let fragments = metadata.fragments else { return metadata.title }
-        return fragments.map { token in
-            return token.text
-        }.joined()
+        let fragments: [DeclarationRenderSection.Token]?
+        
+        if identifier.sourceLanguage == .swift {
```

Comment:
I think we should include a check here to make sure `metadata.navigatorTitle` isn't `nil` just to be extra-safe that we're not breaking any clients of DocC's index generation.

```suggestion
        if identifier.sourceLanguage == .swift || (metadata.navigatorTitle?.isEmpty ?? true) {
```

Reply:
Yep, that's a good call. Fixed

Reply:
Yep, that's a good call. Fixed